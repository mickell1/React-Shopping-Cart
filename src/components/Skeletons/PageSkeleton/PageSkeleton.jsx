import React from 'react'
import ContentLoader from 'react-content-loader'

import styles from './PageSkeleton.scss'

const PageSkeleton = () => (
  <div className={`page-container ${styles['page-skeleton']}`}>
    <h1 className={`h1 ${styles.heading}`}>
      <span className={styles.text}>
        <ContentLoader height={35} width={250} speed={1} primaryColor="#f3f3f3" secondaryColor="#ecebeb" uniquekey="page_header">
          <rect x="0" y="0" rx="2" ry="2" width="250" height="35" />
        </ContentLoader>
      </span>
    </h1>
    <div className="main-section">
      <ContentLoader height={100} width={400} speed={1} primaryColor="#ffffff" secondaryColor="#f5f5f5" uniquekey="page_content">
        <rect x="0" y="0" rx="2" ry="2" width="400" height="100" />
      </ContentLoader>
    </div>
  </div>
)

export default PageSkeleton
