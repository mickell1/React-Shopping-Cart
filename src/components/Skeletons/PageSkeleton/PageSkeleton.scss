.page-skeleton {
  .heading {
    @media (min-width: 768px) {
      text-align: left;
    }
  }

  .text {
    display: inline-block;
    width: 250px;
  }
}
