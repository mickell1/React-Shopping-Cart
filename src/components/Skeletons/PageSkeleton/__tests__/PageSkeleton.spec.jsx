import React from 'react'
import { shallow } from 'enzyme'
import PageSkeleton from '../PageSkeleton'

const setup = () => {
  const component = shallow(<PageSkeleton />)
  return {
    getInstance: () => component,
    getPageSkeleton: () => component.find(PageSkeleton),
  }
}

describe('PageSkeleton', () => {
  it('should render', () => {
    const component = setup()
    expect(component).toBeDefined()
    expect(component).toMatchSnapshot()
  })
})
