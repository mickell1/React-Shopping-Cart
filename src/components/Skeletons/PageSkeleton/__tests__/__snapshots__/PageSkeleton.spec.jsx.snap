// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSkeleton should render 1`] = `
Object {
  "getInstance": [Function],
  "getPageSkeleton": [Function],
}
`;
